devimint/src/tests.rs

```diff
@@ -1812,6 +1813,142 @@ pub async fn recoverytool_test(dev_fed: DevFed) -> Result<()> {
     Ok(())
 }
 
+pub async fn guardian_backup_test(dev_fed: DevFed, process_mgr: &ProcessManager) -> Result<()> {
+    use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_WALLET;
+
+    log_binary_versions().await?;
+
+    let fedimint_cli_version = crate::util::FedimintCli::version_or_default().await;
+    let fedimintd_version = crate::util::FedimintdCmd::version_or_default().await;
+    if VersionReq::parse("<0.3.0-alpha")?.matches(&fedimint_cli_version)
+        || VersionReq::parse("<0.3.0-alpha")?.matches(&fedimintd_version)
+    {
+        info!("Guardian backups didn't exist pre-0.3.0, so can't be tested, exiting");
+        return Ok(());
+    }
+
+    #[allow(unused_variables)]
+    let DevFed {
+        bitcoind,
+        cln,
+        lnd,
+        mut fed,
+        gw_cln,
+        gw_lnd,
+        electrs,
+        esplora,
+    } = dev_fed;
+
+    const PEER_TO_TEST: usize = 0;
+
+    fed.await_all_peers()
+        .await
+        .expect("Awaiting federation coming online failed");
+
+    let client = fed.new_joined_client("guardian-client").await?;
+    let old_block_count = cmd!(
+        client,
+        "dev",
+        "api",
+        "--peer-id",
+        PEER_TO_TEST.to_string(),
+        "module_{LEGACY_HARDCODED_INSTANCE_ID_WALLET}_block_count",
+    )
+    .out_json()
+    .await?["value"]
+        .as_u64()
+        .expect("No block height returned");
+
+    let backup_res = cmd!(
+        client,
+        "--our-id",
+        PEER_TO_TEST.to_string(),
+        "--password",
+        "pass",
+        "admin",
+        "guardian-config-backup"
+    )
+    .out_json()
+    .await?;
+    let backup_hex = backup_res["tar_archive_bytes"]
+        .as_str()
+        .expect("expected hex string");
+    let backup_tar = hex::decode(backup_hex).expect("invalid hex");
+
+    let data_dir = fed
+        .vars
+        .get(&PEER_TO_TEST)
+        .expect("peer not found")
+        .FM_DATA_DIR
+        .clone();
+
+    fed.terminate_server(PEER_TO_TEST)
+        .await
+        .expect("could not terminate fedimintd");
+
+    std::fs::remove_dir_all(&data_dir).expect("error deleting old datadir");
+    std::fs::create_dir(&data_dir).expect("error creating new datadir");
+
+    let write_file = |name: &str, data: &[u8]| {
+        let mut file = std::fs::File::options()
+            .write(true)
+            .create(true)
+            .open(data_dir.join(name))
+            .expect("could not open file");
+        file.write_all(data).expect("could not write file");
+        file.flush().expect("could not flush file");
+    };
+
+    write_file("backup.tar", &backup_tar);
+    write_file("password.private", "pass".as_bytes());
```

Comment:
[nit] It might be more obvious how this filename is special using the pub const

```
write_file(
    fedimint_server::config::io::PLAINTEXT_PASSWORD,
    "pass".as_bytes(),
);
```